packages/workflow/src/internals.ts

```diff
@@ -525,12 +527,21 @@ async function failQuery(queryId: string, error: any) {
   });
 }
 
-export function consumeCompletion(type: keyof State['completions'], taskSeq: number): Completion {
+/** Consume a completion if it exists in Workflow state */
+export function maybeConsumeCompletion(type: keyof State['completions'], taskSeq: number): Completion | undefined {
   const completion = state.completions[type].get(taskSeq);
+  if (completion !== undefined) {
+    state.completions[type].delete(taskSeq);
+  }
+  return completion;
+}
+
+/** Consume a completion if it exists in Workflow state, throws if it doesn't */
+export function consumeCompletion(type: keyof State['completions'], taskSeq: number): Completion {
+  const completion = maybeConsumeCompletion(type, taskSeq);
   if (completion === undefined) {
     throw new IllegalStateError(`No completion for taskSeq ${taskSeq}`);
```

Comment:
> I think we should change that message too. "Completion" is a confusing word there. It's more like "No pending awaitable for...", no?

Now, are there any cases in which a user would see this error?

Reply:
A user could see this error if new workflow code processes history generated by old code.